File: 算法和数据结构基础/3.1 暴力美学-排序/code/java/SortDemo1.java
//SortDemo1.java
import java.io.*;
import java.util.*;


public class SortDemo1
{
    protected static final String progName = "SortDemo1";

    /**可以调用的算法*/
    public enum Mode {
        BUB,
        SELECT
    }

    /**
     打印基本信息。
     */
    protected static void printUsage(String progName) {
        System.err.println("USAGE: " + progName + " [sort method] [input file]");
        System.err.println("  sort methods [bubble, quick, merge, cocktail]");
        System.err.println("EXAMPLE: " + progName + " quick random.txt");
    }
    /**
     * 主函数。接受命令行参数-包括算法名字和排序数据集
     */
    public static void main(String[] args) {
        try {
            if (args.length != 2) {
                printUsage(progName);
                System.exit(1);
            }
            // 获得排序算法名
            String algorithmUsed = args[0];
            // 获得需要排序的数据集
            String fileName = args[1];

            File inFile = new File(fileName);
            Scanner in = new Scanner(inFile);

            // 从文件读入数据
            ArrayList<Integer> buffer = new ArrayList<Integer>();
            while (in.hasNextInt()) {
                buffer.add(new Integer(in.nextInt()));
            }

            // 使用读入的数据创建列表
            int[] array = new int[buffer.size()];
            Iterator bit = buffer.iterator();
            int j = 0;
            while (bit.hasNext()) {
                array[j] = (Integer) bit.next();
                j++;
            }
            buffer = null;

            // 选择合适的算法
            SortAlgorithm sortAlgor = null;
            switch(algorithmUsed) {
                case "bubble":
                    sortAlgor = new BubbleSort();
                    break;
                case "select":
                    sortAlgor = new SelectionSort();
                    break;
                default:
                    System.err.println("Error: " + algorithmUsed + "is invalid.");
                    array = null;
                    printUsage(progName);
                    System.exit(1);
            }

            long startTime = System.nanoTime();
            // 排序
            sortAlgor.sort(array);
            long endTime = System.nanoTime();
            // 打印排序后的数组
//            for (int i = 0; i < array.length; i++) {
//                System.out.println(array[i]);
//            }
            double timeElapsed = (endTime - startTime) / Math.pow(10, 9);
            System.out.println("Time elapsed (secs): " + timeElapsed);
        }
        catch (Exception e) {
            System.err.println(e.getMessage());
            printUsage(progName);
        }
    }
} 

File: 算法和数据结构基础/3.1 暴力美学-排序/code/python/SortDemo1.py
import argparse
import sys
import time
from BubbleSort import BubbleSort
from SelectionSort import SelectionSort


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument(
        '--method',
        type=str,
        default='bubble',
        help="""
                Method of sort
            """
    )

    parser.add_argument(
        '--data',
        type=str,
        default='./debug.txt',
        help="""
            Path to data file
        """
    )

    args = parser.parse_args()
    #获得排序算法名
    algorithmUsed = args.method
    #获得需要排序的数据集
    fileName = args.data;            
    # 从文件读入数据,并使用读入的数据创建列表
    array=[]
    with open(fileName,'r') as f:
        for line in f:
            temp1 = line.strip('\n')
            temp2 = temp1.strip('\t')
            array.append(temp2)
            k = len(array)
        for i in range(len(array)):
            array.append(float(array[i]))
        del(array[0:k])
    # 选择合适的算法
    if algorithmUsed == 'bubble':
        sortAlgor = BubbleSort()
    elif algorithmUsed == 'select':
        sortAlgor = SelectionSort()
    else:
        print("Error: " + algorithmUsed + "is invalid.")
        array = None
        sys.exit(1)
    time_start=time.time()
    #排序
    array = sortAlgor.mysort(array)
    #length = len(array)
    #for num in range(length):
    #    print(array[num])
    time_end=time.time()
    print('Time cost:',time_end-time_start,'s')